Implement button visualizer

MoodyMan04 opened the pull request:
basic implementation of button class for visualizer
- not ready for merge yet
- includes some testing in adapter class
- mouse_hover seems to work, struggling on mouse_click and event handling

JeanVArdent committed:
Fix merge (#46)

* Fix Merge

KingPhilip14 committed:
Fixed README (#68)

Added more info for the avatar class in the README.

MoodyMan04 committed:
Merge branch 'master' into implement-button-visualizer

MoodyMan04 committed:
Merge branch 'Visualizer' into implement-button-visualizer

# Conflicts:
#	README.md

MoodyMan04 committed:
basic button implementation

- still needs work
- includes some testing in adapter class

MoodyMan04 committed:
fixes and changes

added:
- better documentation
- background button to text
- padding parameter
- border_radius parameter (to round corners of button
- color parameters for bg button
- changed collision logic to work with bg rect, not text rect

MoodyMan04 committed:
Merge branch 'Visualizer' into implement-button-visualizer

JeanVArdent commented on visualizer/adapter.py:
```diff
@@ -27,8 +39,11 @@ def populate_bytesprites(self) -> pygame.sprite.Group:
         self.populate_bytesprite.add(ExampleBS(self.screen))
         return self.populate_bytesprite.copy()
 
-    def render(self, sidebars: Sidebars) -> None:
```
Still need the sidebars, reincorporate this please

JeanVArdent commented on visualizer/utils/button.py:
```diff
@@ -0,0 +1,220 @@
+border_radius      :  Level of smoothing to corners of button       Default - 5
+
+In future projects, defaults for button style should be changed according to style of game for ease of code
+"""
```
Good to see this!!

JeanVArdent commented on visualizer/utils/button.py:
```diff
@@ -0,0 +1,220 @@
+
+
+class Button(Text):
+    def __init__(self, screen: pygame.Surface, text: str, action: Callable,
```
I am going to go back on what I said; create a class to hold the 6 colors and extract the logic for the getters and setters there then use that class as a parameter for this, there are too many elements in this constructor.

JeanVArdent reviewed (changes requested):
Good direction, a few changes that are needed. Also expand the documentation for the text class please. The class documentation should be under the class definition as well, otherwise it won't be picked up on autodoc

MoodyMan04 committed:
code clean up + timer

cleaned up code and added timer in button class to help display clicked colors better

MoodyMan04 committed:
fixed documentation + timer

- relocated documentation to proper location
- fixed button timer to be adjustable and consistent with frame rate

MoodyMan04 committed:
more timer changes

- reverted back to milliseconds, which do not need to be multiplied by the frame rate
- used math.floor when needed
- changed click_duration to int

JeanVArdent reviewed (approved):
Lgtm

KingPhilip14 commented on visualizer/utils/button.py:
```diff
@@ -0,0 +1,277 @@
+    Can select three colors for both the text and button: default, mouse hover, and mouse clicked
+
+    Parameters:
+    fg_color           :  Used to store default text color              Default - #daa520
```
It's good that the hex codes are here but add the actual color names next to the defaults to make it easier

KingPhilip14 commented on visualizer/utils/button.py:
```diff
@@ -0,0 +1,277 @@
+
+class Button(Text):
+    """
+    Class that creates an intractable button extending the Text class.
```
intractable or interactable?

KingPhilip14 reviewed (changes requested):
Just a couple comments. Otherwise, it looks good. I could make most sense of it because of the comments. Good job!

The-8th-wonder reviewed (approved):
LGTM👍😱

KingPhilip14 reviewed (approved):
Never mind, looks good (as per Jean's request).
